### PR TITLE
BLE central Role implementation and peripheral refactor

### PR DESCRIPTION
**Pull request by sgbihu**
Need update the BLE's FW first.  @kevin306 has build this FW.
@SidLeung

### CODE REVIEW

File: libraries/CurieBLE/examples/LEDCentral/LEDCentral.ino [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+bool adv_found(uint8_t type, 
+               const uint8_t *data, 
+               uint8_t data_len,
+               void *user_data)
+{
+    bt_addr_le_t *addr = (bt_addr_le_t *)user_data;
+    int i;
+
+    pr_debug(LOG_MODULE_APP, "[AD]: %u data_len %u\n", type, data_len);
```
Comment (eriknyquist) [inline libraries/CurieBLE/examples/LEDCentral/LEDCentral.ino]: Can we replace all calls to pr_debug with Serial.println, please?

Comment (SidLeung) [inline libraries/CurieBLE/examples/LEDCentral/LEDCentral.ino]: Ok...  We need to distinguish the difference between pr_debug and Serial.print.  Somehow, need to make user to understand that pr_debug is for debugging purposes that log messages go out to another serial port.  Serial.print is the standard IO serial output for most users.  Any suggestions on how to make that clear?  May be a comment in the code?  That would be a whole lot of comments.  May be rename the routine a bit more obvious?

Comment (sgbihu) [inline libraries/CurieBLE/examples/LEDCentral/LEDCentral.ino]: In the example sketches. I think we didn"t expose the pr_debug interface. I modified all pr_debug in example sketch.

**Comment by eriknyquist** [comment]
Hi @sgbihu, can you please clean up the history? Remove the merge commit, and please make one commit for each change, e.g. one for BLE central implementation and one for the peripheral refactor.
Also, when you are writing the commit messages, please consider our [commit message guidelines](https://github.com/01org/corelibs-arduino101/wiki/Writing-a-commit-message).

**Comment by calvinatintel** [comment]
@sgbihu Please crush the commits into one and rebase to the top of master. The branch has conflicts and cannot be merged at this point.

**Pull request closed without merging**